lib/vrl/stdlib/src/parse_key_value.rs

```diff
@@ -98,7 +98,7 @@ impl Expression for ParseKeyValueFn {
     }
 
     fn type_def(&self, _: &state::Compiler) -> TypeDef {
-        TypeDef::new().infallible().object::<(), Kind>(map! {
+        TypeDef::new().fallible().object::<(), Kind>(map! {
```

Comment:
This change seems unrelated and also might not be correct, as it can fail at runtime e.g. when the input ended unexpectedly?

Reply:
Yeah, `parse_key_value("ook")` will fail since there is no key=value. I needed a good function that could fail that returned a map to test the merge code, and noticed that the type def here was incorrectly specified, so I just fixed it to use in the tests..

Reply:
Oh, I somehow read the diff the other way around. All good 😄